index.js

```diff
@@ -0,0 +1,60 @@
+
+/**
+ * This strategy is an intermediate example of how to craft and equip items, and harvest blocks with the rg-bot package.
+ * The Bot will create a pickaxe and use it to mine the bell in the starting village
+ *
+ * @param {RGBot} bot
+ */
+function configureBot(bot) {
+
+    bot.setDebug(true);
+
+    // This is our main loop. The Bot will invoke this on spawn.
+    // goal: Gather wood, use it to craft a pickaxe, and then dig the Bell in the starting village.
+    async function startRoutine() {
+
+        // Two of the houses in the starting village have a chest containing logs as well as a crafting table.
+        // These are the perfect places to quickly craft a pickaxe, which will allow the Bot to collect the village's Bell.
+        // First, find and approach the nearest crafting table
+        let craftingTable = await bot.findBlock('crafting_table');
+        const pathingSuccessful = await bot.approachBlock(craftingTable, {reach: 3});
+        if(!pathingSuccessful) {
+            // Sometimes the Bot may encounter pathing issues. Since there are two houses with the perfect
+            // conditions for this strategy, the Bot can try to get to the second house if it fails to approach the first one.
+            craftingTable = await bot.findBlock('crafting_table', { skipClosest: true });
+            await bot.approachBlock(craftingTable, {reach: 3});
+        }
+
+        // Next, loot some logs from the chest in the same house.
+        // The chest has three, but the Bot only needs two to craft the pickaxe,
+        // so we won't let it be too greedy.
+        const chest = bot.findBlock('chest', { maxDistance: 10 });
+        await bot.approachBlock(chest);
+        const chestInventoryWindow = await bot.mineflayer().openContainer(chest);
+        await bot.withdrawItems(chestInventoryWindow, { itemName: 'spruce_log', quantity: 2 });
+        await chestInventoryWindow.close();
+
+        // Craft the components the Bot will need for one pickaxe
+        // Turn the logs into 8 planks, and then two of the planks into some sticks
+        await bot.craftItem('spruce_planks', { quantity: 2 });
+        await bot.craftItem('stick');
+
+        // Now the Bot has enough materials to craft a pickaxe
+        await bot.approachBlock(craftingTable);
+        await bot.craftItem('wooden_pickaxe', { craftingTable: craftingTable });
+        await bot.holdItem('wooden_pickaxe');
+
+        // Finally, have the Bot collect the Bell using its new pickaxe
+        await bot.findAndDigBlock('bell', { maxDistance: 100 });
+        bot.chat(`I have collected the village's Bell!`);
+    }
+
```

Comment:
Can we give this bot a secondary objective after the bell (even if it is poppy farming).

I would like to point to these samples as the first 'system' bots and need them to be somewhat capable of winning 🗡️ 


Also, what happens if the bell has already been taken ?

Reply:
Not sure I'm sold on adding more to this sample because I don't think these need to be complete strategies in order for a new player to try them out and learn how to use rg-bot. If these are the first things people are going to look at as "how to make a bot" then I'd prefer keeping them simple over "here is a huge blob of code to handle movement issues, crafting issues, etc. go slap this right into a multiplayer match"

Reply:
With the way it's written now, then findAndDigBlock won't fail if the bell isn't present (it will return false) and you'd still see the "I have collected the village's Bell!" message in chat even though the bot didn't collect anything